File: Excel/450 Challenge.py
import pandas as pd

input = pd.read_excel("450 Ranking.xlsx",  usecols="A:C", nrows=20)
test = pd.read_excel("450 Ranking.xlsx", usecols="D:D", nrows=20)

input["rank"] = input.groupby("Company")["Sales"].rank(method="dense", ascending=False)

print(input["rank"].equals(test["Answer Expected"])) # True

File: Excel/400-499/450/450 Challenge.R
library(tidyverse)
library(readxl)

input = read_excel("Excel/450 Ranking.xlsx", range = "A1:C20")
test  = read_excel("Excel/450 Ranking.xlsx", range = "D1:D20")

result = input %>%
  mutate(rank = dense_rank(desc(Sales)), .by = Company)

all.equal(result$rank, test$`Answer Expected`)
# [1] TRUE